terraform/install.sh

```diff
@@ -256,28 +258,30 @@ getExternalIp() {
 # Install Load Generator service and start generating synthetic traffic to Sandbox
 loadGen() {
   log "Running load generator"
-  # remove existing load generator
-  old_loadgen=$(gcloud compute instances list --project $project_id --filter="name:loadgenerator*" --format="value(name)")
-  if [[ -n "$old_loadgen" ]]; then
-    loadgen_zone=$(gcloud compute instances list --project $project_id --filter="name:loadgenerator*" --format="value(zone)")
-    gcloud compute instances delete $old_loadgen --zone $loadgen_zone --quiet
-  fi
   # launch a new load generator
-  ../loadgenerator/loadgen autostart $external_ip
+  pushd loadgen/
+  terraform init -lock=false
+  terraform apply --auto-approve -var="project_id=${project_id}" -var="external_ip=${external_ip}"
```

Comment:
loadgen provisioning should be part of terraform provisioning that includes the rest of the cluster. It should not be invoked separately as this way we'll have a dedicated state (.tfstate) and it will hurt our ability to resume cluster creation in case it didn't work for the first time.

Reply:
will work on this - the clusters can be provisioned at the same time, but it'll require a bit of refactoring

Reply:
Given that your change is already pretty big, I suggest splitting it into a separate CR after this one is merged.

Reply:
How much work/time would be involved in making the change?

Reply:
i can just put the cluster-provisioning in the same folder as the hipster shop stuff - that is pretty straightforward and achieves our main goal of having those two tasks run concurrently (saving a lot of time). Moving the rest of the load generator stuff over might get a little complicated.

Reply:
I'd recommend using a module https://www.terraform.io/docs/configuration/modules.html to be able to keep all the loadgen stuff together and you can pass all the variables you need for the module in. You can use a script to pull the external IP and use the external data Terraform resource to retrieve it for your variables. I have an incomplete version here https://github.com/nicklin37/stackdriver-sandbox/blob/monitoring_integration/terraform/04_monitoring.tf

Reply:
+1 to Nic's suggestion. We can do it in a new PR though after this one is merged.